File: fall-2022/cs1c/HW/HW04/src/models/Address.cpp
#include "Address.h"

Address &Address::operator=(const Address &rhs) {
  if (this == &rhs) {
    return *this;
  }

  this->setStreet(rhs.street_);
  this->setCity(rhs.city_);
  this->setState(rhs.state_);
  this->setZip(rhs.zip_);

  return *this;
}

std::string Address::getFullAddress() const {
  return this->street_ + " " + this->city_ + " " + this->state_ + " " +
         std::to_string(this->zip_);
}

void Address::setStreet(const std::string &street) {
  this->street_ = street;
}

std::string Address::getStreet() const {
  return this->street_;
}

void Address::setCity(const std::string &city) {
  this->city_ = city;
}

std::string Address::getCity() const {
  return this->city_;
}

void Address::setState(const std::string &state) {
  this->state_ = state;
}

std::string Address::getState() const {
  return this->state_;
}

void Address::setZip(const int &zip) {
  this->zip_ = zip;
}

int Address::getZip() const {
  return this->zip_;
}

void Address::print() const {
  std::cout << "Address: " << this->getFullAddress() << "\n";
}

File: spring-2023/cs4b/final/src/CourseManager/models/academics/Address.java
package CourseManager.models.academics;

public class Address {
  public Address() {
  }

  public Address(String street, String city, String state, int zip) {
    m_street = street;
    m_city = city;
    m_state = state;
    m_zip = zip;
  }

  public String getStreet() {
    return m_street;
  }

  public void setStreet(String street) {
    m_street = street;
  }

  public String getCity() {
    return m_city;
  }

  public void setCity(String city) {
    m_city = city;
  }

  public String getState() {
    return m_state;
  }

  public void setState(String state) {
    m_state = state;
  }

  public int getZip() {
    return m_zip;
  }

  public void setZip(int zip) {
    m_zip = zip;
  }

  @Override
  public String toString() {
    return "ADDRESS: " + m_street + " " + m_city + " " + m_state + " " + Integer.toString(m_zip);
  }

  private String m_street;
  private String m_city;
  private String m_state;
  private int m_zip;
}
